app/models/manageiq/providers/vmware/infra_manager/inventory/collector/property_collector.rb

```diff
@@ -79,6 +79,11 @@ def process_prop_change(prop_hash, prop_change)
     h, prop_str = hash_target(prop_hash, prop_change.name)
     tag, key    = tag_and_key(prop_str)
 
+    if h.nil?
+      _log.warn("Invalid property change #{prop_change.name}")
+      return
```

Comment:
I'm a little wary of doing this, mainly because it is impossible to tell the difference between a bug in our cache code vs a bug on the vSphere side (like this case IMO).

Reply:
For reference, `h` here is what would be at `host["host-1234"]` and `prop_str` would be (in this case) `:mountInfo`